meta-facebook/gt-cc/src/platform/plat_hook.c

```diff
@@ -990,6 +992,42 @@ bool post_ltc4282_read(uint8_t sensor_num, void *args, int *reading)
 	return true;
 }
 
+bool post_ltc4286_read(uint8_t sensor_num, void *args, int *reading)
+{
+	if (!reading) {
+		post_i2c_bus_read(sensor_num, args, reading);
+		return false;
+	}
+	ARG_UNUSED(args);
+
+	if (!post_i2c_bus_read(sensor_num, args, reading))
+		return false;
+
+	sensor_cfg *cfg = &sensor_config[sensor_config_index_map[sensor_num]];
+
+	sensor_val *sval = (sensor_val *)reading;
+	float val = sval->integer + (sval->fraction * 0.001);
+
+	/* Adjust the power value for accuracy by power team requirement */
+	switch (cfg->offset) {
+	case PMBUS_READ_VOUT:
+		return true;
+	case PMBUS_READ_IOUT:
+		val = ADJUST_LTC4286_CURRENT(val);
+		break;
+	case PMBUS_READ_PIN:
+		val = ADJUST_LTC4286_POWER(val);
+		break;
+	default:
+		return false;
+	}
+
+	sval->integer = (int16_t)val;
+	sval->fraction = (val - sval->integer) * 1000;
```

Comment:
Are these two lines of code expected to do anything?  It seems they set values but as soon as the function returns, they are out of scope.

Reply:
Hi @GoldenBug, 
The sensor_val format is a two-byte integer value and two bytes fraction value, we change it to the float value for calculating the value from the formula then reassembly to the sensor_val format.

Thanks.